### DISCUSSION TITLE
Having issues in coming up with schema

### DISCUSSION THREAD

**Discussion by Ranjeetk489** [prisma/prisma#16002] [Q&A]
Hi, I am new to prisma and what i want to create is a inventory model which contains references to other items model for eg food, water etc. For that purpose I'm trying out this model. I tried to perform crud on this but i am not getting the result i want to see. What i want is to have inventory auto filled with the data from water and food model. It'd really helpful if someone could point out what i am doing wrong or any references. Thanks

```
model inventory {
  id    String  @id() @default(cuid())
  water water[]
  food  food[]
}

model water {
  id          String     @id
  type        String?    @default("Water")
  inventory   inventory? @relation(fields: [inventoryId], references: [id])
  inventoryId String?
  quantity    Int
}

model food {
  id             String     @id
  type           String?    @default("Food")
  packet_content String
  calories       Int
  expiry         DateTime
  inventory      inventory? @relation(fields: [inventoryId], references: [id])
  inventoryId    String?
```

**Comment by Rackar** [comment]
the schema is ok. when you add water record, you must give the exist inventoryId to link

**Comment by pantharshit00** [comment]
> What i want is to have inventory auto filled with the data from water and food model.

Do you mean here as soon as you create an inventory, it should have some food and water or you mean each water or food item should get assigned an inventory automatically?

**Reply by vladi-stevanovic** [reply]
:wave: Hello @Ranjeetk489 , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
